src/XMakeBuildEngine/Microsoft.Build.csproj

```diff
@@ -293,8 +293,8 @@
     <Compile Include="Construction\Solution\SolutionFile.cs">
       <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
     </Compile>
-    <Compile Include="Debugger\DebuggerLocalType.cs" />
-    <Compile Include="Debugger\DebuggerManager.cs" />
+    <Compile Condition="'$(MsbuildDebugger)' == 'true'" Include="Debugger\DebuggerLocalType.cs" />
+    <Compile Condition="'$(MsbuildDebugger)' == 'true'" Include="Debugger\DebuggerManager.cs" />
     <!-- #### DEFINITION MODEL ### -->
     <Compile Include="Definition\BuiltInMetadata.cs" />
     <Compile Include="Definition\ProjectCollection.cs" />
```

Comment:
What happens in VS with a condition on the Compile? Does it show in the UI? I still think I would prefer to see #if in the file rather than this (same feedback on @dsplaisted latest PR). If we think this is a good idea though that's fine.